From 14f92b1479aa065edf3f0aa86b87d4c4ff1fe2ba Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Wed, 6 Feb 2013 23:14:22 -0500
Subject: [PATCH] Fix media-keys handling with GTK+ 3.7.8

GDK now requests XI 2.3 and if the X server supports that
version, subsequent requests for a lower version yield
a BadValue. Therefore, we need to change the code in
gsd-input-helper.c to request 2.3 as well.
---
 plugins/common/gsd-input-helper.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

Index: b/plugins/common/gsd-input-helper.c
===================================================================
--- a/plugins/common/gsd-input-helper.c
+++ b/plugins/common/gsd-input-helper.c
@@ -127,18 +127,11 @@
         gdk_error_trap_push ();
 
         major = 2;
-        minor = 0;
+        minor = 3;
 
         if (XIQueryVersion (GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()), &major, &minor) != Success) {
                 gdk_error_trap_pop_ignored ();
-                /* try for 2.2, maybe gtk has already announced 2.2 support */
-                gdk_error_trap_push ();
-                major = 2;
-                minor = 2;
-                if (XIQueryVersion (GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()), &major, &minor) != Success) {
-                    gdk_error_trap_pop_ignored ();
                     return FALSE;
-                }
         }
         gdk_error_trap_pop_ignored ();
 
